x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/data_stream/list_page/components/data_stream_row_actions.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo } from 'react';
+
+import { FormattedMessage } from '@kbn/i18n/react';
+import { useKibanaLink } from '../../../../hooks/use_kibana_link';
+import { DataStream } from '../../../../types';
+import { TableRowActionsNested } from '../../../../components/table_row_actions_nested';
+
+export const DataStreamRowActions = memo<{ datastream: DataStream }>(({ datastream }) => {
+  const { dashboards } = datastream;
+  const panels = [];
+  const actionNameSingular = (
+    <FormattedMessage
+      id="xpack.ingestManager.dataStreamList.viewDashboardActionText"
+      defaultMessage="View dashboard"
+    />
+  );
+  const actionNamePlural = (
+    <FormattedMessage
+      id="xpack.ingestManager.dataStreamList.viewDashboardsActionText"
+      defaultMessage="View dashboards"
+    />
+  );
+
+  if (!dashboards || dashboards.length === 0) {
+    panels.push({
+      id: 0,
+      items: [
+        {
+          icon: 'dashboardApp',
+          disabled: true,
+          name: actionNameSingular,
+        },
+      ],
+    });
+  } else if (dashboards.length === 1) {
+    panels.push({
+      id: 0,
+      items: [
+        {
+          icon: 'dashboardApp',
+          href: useKibanaLink(`/dashboard/${dashboards[0].id || ''}`),
+          name: actionNameSingular,
+        },
+      ],
+    });
+  } else {
+    panels.push({
+      id: 0,
+      items: [
+        {
+          icon: 'dashboardApp',
+          panel: 1,
+          name: actionNamePlural,
+        },
+      ],
+    });
+    panels.push({
+      id: 1,
+      title: 'View dashboards',
```

Comment:
This gave a type error from somewhere within EUI when I used a `<FormattedMessage />` instead, so this is not translated. I'm sure there's a way to fix that, comments welcome.

Reply:
maybe use i18n.translate instead of FormattedMessage? the prop might expect a string instead of react node